plugins/cloudbuild/src/components/Cards/Cards.tsx

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React, { useEffect } from 'react';
+import { useWorkflowRuns } from '../useWorkflowRuns';
+import { WorkflowRun, WorkflowRunsTable } from '../WorkflowRunsTable';
+import { Entity } from '@backstage/catalog-model';
+import { WorkflowRunStatus } from '../WorkflowRunStatus';
+import {
+  Link,
+  Theme,
+  makeStyles,
+  LinearProgress,
+  Typography,
+} from '@material-ui/core';
+import {
+  InfoCard,
+  StructuredMetadataTable,
+  errorApiRef,
+  useApi,
+} from '@backstage/core';
+import ExternalLinkIcon from '@material-ui/icons/Launch';
+import { CLOUDBUILD_ANNOTATION } from '../useProjectName';
+
+const useStyles = makeStyles<Theme>({
+  externalLinkIcon: {
+    fontSize: 'inherit',
+    verticalAlign: 'bottom',
+  },
+});
+
+const WidgetContent = ({
+  error,
+  loading,
+  lastRun,
+  branch,
+}: {
+  error?: Error;
+  loading?: boolean;
+  lastRun: WorkflowRun;
+  branch: string;
+}) => {
+  const classes = useStyles();
+  if (error) return <Typography>Couldn't fetch latest {branch} run</Typography>;
+  if (loading) return <LinearProgress />;
+  return (
+    <StructuredMetadataTable
+      metadata={{
+        status: (
+          <>
+            <WorkflowRunStatus status={lastRun.status} />
+          </>
+        ),
+        message: lastRun.message,
+        url: (
+          <Link href={lastRun.googleUrl} target="_blank">
+            See more on Google{' '}
+            <ExternalLinkIcon className={classes.externalLinkIcon} />
+          </Link>
+        ),
+      }}
+    />
+  );
+};
+
+export const LatestWorkflowRunCard = ({
+  entity,
+  branch = 'master',
+}: {
+  entity: Entity;
+  branch: string;
+}) => {
+  const errorApi = useApi(errorApiRef);
+  const projectId = entity?.metadata.annotations?.[CLOUDBUILD_ANNOTATION] || '';
+
+  const [{ runs, loading, error }] = useWorkflowRuns({
+    projectId,
+  });
+  const lastRun = runs?.[0] ?? ({} as WorkflowRun);
+  useEffect(() => {
+    if (error) {
+      errorApi.post(error);
+    }
+  }, [error, errorApi]);
+
+  return (
+    <InfoCard title={`Last ${branch} build`}>
+      <WidgetContent
+        error={error}
```

Comment:
oh hey i see now that you use both the errorApi AND a text output for the same error - maybe best to decide which one of them to use - could be reasonable to inside the InfoCard have a switch that if there's an error draw a WarningPanel, if it's loading draw a progress bar, otherwise draw WidgetContent.

That way, WidgetContent does not have to have neither know of loading nor error states. And then you can remove the errorApi usage here

Reply:
This I copied from the github-actions plugin. You can see it here: https://github.com/spotify/backstage/blob/17c30eaf3efd026032110c8d80217e63f4385b9f/plugins/github-actions/src/components/Cards/Cards.tsx#L104 
I used this plugin as a base for writing the google Cloud Build plugin.